src/frequenz/channels/utils/file_watcher.py

```diff
@@ -62,19 +64,35 @@ def __del__(self) -> None:
         """
         self._stop_event.set()
 
-    async def receive(self) -> Optional[pathlib.Path]:
+    async def ready(self) -> None:
         """Wait for the next file event and return its path.
 
+        Raises:
+            StopAsyncIteration: When the channel is closed.
+
         Returns:
             Path of next file.
         """
-        while True:
-            changes = await self._awatch.__anext__()
-            for change in changes:
-                # Tuple of (Change, path) returned by watchfiles
-                if change is None or len(change) != 2:
-                    return None
-
-                _, path_str = change
-                path = pathlib.Path(path_str)
-                return path
+        # if there are messages waiting to be consumed, return immediately.
+        if self._changes:
+            return
+
+        self._changes = await self._awatch.__anext__()
+
+    def consume(self) -> pathlib.Path:
+        """Return the latest change once `ready` is complete.
+
+        Raises:
+            ChannelClosedError: When the channel is closed.
+
+        Returns:
+            The next change that was received.
+        """
+        assert self._changes, "calls to `consume()` must be follow a call to `ready()`"
+        change = self._changes.pop()
+        # Tuple of (Change, path) returned by watchfiles
+        if change is None or len(change) != 2:
+            raise ChannelClosedError()
```

Comment:
This is also weird, but pre-existing, so I created https://github.com/frequenz-floss/frequenz-channels-python/issues/54.